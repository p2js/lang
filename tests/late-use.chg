use x as f1;

use f3 as f4;

fn x() {}

use x as f2;

use f1 as f3;

use f4 as f5;

fn main() {
    f1();
    f2();
    f3();
    f6();
    f4();
    f5();
    f6();
}

use f5 as f6;
