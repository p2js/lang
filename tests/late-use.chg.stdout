expr @ 5:8 6:0 => ()
expr @ 14:5 14:7 => fn x(...)
expr @ 14:5 14:9 => ()
expr @ 15:5 15:7 => fn x(...)
expr @ 15:5 15:9 => ()
expr @ 16:5 16:7 => fn x(...)
expr @ 16:5 16:9 => ()
expr @ 17:5 17:7 => fn x(...)
expr @ 17:5 17:9 => ()
expr @ 18:5 18:7 => fn x(...)
expr @ 18:5 18:9 => ()
expr @ 19:5 19:7 => fn x(...)
expr @ 19:5 19:9 => ()
expr @ 20:5 20:7 => fn x(...)
expr @ 20:5 20:9 => ()
expr @ 13:11 22:0 => ()



define void @root$f6() {
prologue:
br label %bb.0

bb.0:
ret void

}

define void @main() {
prologue:
br label %bb.0

bb.0:
%l.0 = call {} bitcast(void()* @root$f6 to {}()*)()
br label %bb.1

bb.1:
%l.1 = call {} bitcast(void()* @root$f6 to {}()*)()
br label %bb.2

bb.2:
%l.2 = call {} bitcast(void()* @root$f6 to {}()*)()
br label %bb.3

bb.3:
%l.3 = call {} bitcast(void()* @root$f6 to {}()*)()
br label %bb.4

bb.4:
%l.4 = call {} bitcast(void()* @root$f6 to {}()*)()
br label %bb.5

bb.5:
%l.5 = call {} bitcast(void()* @root$f6 to {}()*)()
br label %bb.6

bb.6:
%l.6 = call {} bitcast(void()* @root$f6 to {}()*)()
br label %bb.7

bb.7:
ret void

}


